src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -1028,6 +1028,8 @@ private void AddOrReplacePeerTip(int networkPeerId, uint256 newTip)
 
             this.peerTipsByPeerId.Add(networkPeerId, newTip);
 
+            this.chainState.IsAtBestChainTip = this.IsAtBestChainTip();
```

Comment:
I think this does not belong here, this should be in consensus manager. It is missing there as we only have eight in a single place – when consensus tip changes and it is missing when the peer presents new header

Reply:
It's already in CM in ConnectBlockAsync after we call CHT.FullValidationSucceeded